File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        difftarget: Dict = {}
        
        for index, value in enumerate(nums):
            if value in difftarget.keys():
                return [index, difftarget[value]]
            else:
                difftarget[target-value] = index
        
            
            

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int, int> diffTarget;
        vector<int> returnValue;
        
        for(int i=0; i<nums.size(); i++){
            
            map<int,int>::iterator it = diffTarget.find(nums[i]);
            if(it != diffTarget.end())
            {
                returnValue = {it->second, i};
                return returnValue;
            }
            
            diffTarget.insert(pair<int, int>(target-nums[i], i));
        
        }
        return returnValue;
    }
};